### PR TITLE
[DISCUSS] Introduce Checkstyle for code style checks

### PR DESCRIPTION
**Pull request by mxm**
This contains two commits which introduce a) the Checkstyle plugin with a limited rule set to our Gradle build. b) code adjustments to comply with the rule set. Checkstyle checks run after compiling and are only applied for the main source code (no testing code checks). They are enforced because otherwise they are practicably useless (looking at the JavaDoc warnings..). The checks can be easily also integrated into IntelliJ.

Before this gets reviewed, I want to get some feedback from the CrateDB developers. Firstly, I'd like to know if they like to have to Checkstyle. Secondly, the rule set is by no means final. I've lifted the rule set quite a bit from the original Google style to require less code changes. For example, we might want to enforce a line limit of 120 or so, but I left that open for now. Also, we might want to check testing code as well in the future.

Let me know what you think.

### CODE REVIEW

File: sql/src/main/java/io/crate/operation/reference/sys/node/fs/NodeStatsFsDataExpression.java [hunk 1]
```diff
@@ -43,10 +43,12 @@ public NodeStatsFsDataExpression() {
 
     @Override
     protected Map<String, Object> valueForItem(final ExtendedFsStats.Info input) {
-        return new HashMap<String, Object>() {{
-            put(NodeFsStatsExpression.DEV, input.dev());
-            put(NodeFsStatsExpression.PATH, input.path());
-        }};
+        return new HashMap<String, Object>() {
+            {
+                put(NodeFsStatsExpression.DEV, input.dev());
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/reference/sys/node/fs/NodeStatsFsDataExpression.java]: Not relevant for this PR, but I think we should change this here to not create anynomous HashMap classes

**Comment by mxm** [comment]
Great! Sorry, the test failures are fixed.

**Comment by andreidan** [comment]
Looks good! 
What do you think about adding the declaration order check as well?
 (we normally try to obey this anyhow) http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/checks/coding/DeclarationOrderCheck.html

**Comment by mxm** [comment]
I've taken a look at the line length and I see a lot of cases where we would increase readability by limiting the line length. Just to give an example:

On `master`:
screen shot 2017-09-08 at 11 35 47
With line length limit:
screen shot 2017-09-08 at 11 36 55

I suppose if started limiting the line we also would have to check for indention style used for parameter types, e.g. here it is the IntelliJ style.

File: sql/src/main/java/io/crate/operation/reference/sys/node/fs/NodeStatsFsDataExpression.java (hunk 1 above)
Comment (matriv) [reply sql/src/main/java/io/crate/operation/reference/sys/node/fs/NodeStatsFsDataExpression.java]: @mxm check also here: https://blog.jooq.org/2014/12/08/dont-be-clever-the-double-curly-braces-anti-pattern/

**Comment by mxm** [comment]
@andreidan Good idea. Just tried it out. This could be a bit more invasive change at it touches many files and also reorders their contents (methods moving around): "Checkstyle found 543 item(s) in 225 file(s)" https://pastebin.com/LK2UVN34

**Comment by matriv** [comment]
@mxm Looks good! Imho I wouldn't strictly enforce the line length threshold.

**Comment by matriv** [comment]
@mxm Yeah, I'd say not to do apply the DeclarationOrder, just the ModifierOrder which you've already applied.

**Comment by mxm** [comment]
Rebased the PR to the latest master after #6256 had been merged. Will merge later if there are no further comments.

**Comment by matriv** [comment]
@mxm don't forget to squash last commit (changes after rebase)

**Comment by mxm** [comment]
@matriv I will squash all the commits (=just have one commit) since introducing the Checkstyle and then doing the style changes would break the first commit. I split up the commits to ease reviewing.

**Pull request merged**